### PR TITLE
feat: add `afterHashbangComment` option to `lines-around-comment` rule

### LINKED ISSUE #16913: Rule Change: add `afterHashbangComment` option to `lines-around-comment`
### What rule do you want to change?

`lines-around-comment`

### What change to do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
#!foo
anyStatement;
```

### What does the rule currently do for this code?

Does nothing

### What will the rule do after it's changed?

Warns `Expected line after comment` with following options:

```
"lines-around-comment": ["error", { "afterHashbangComment": true }]
```

### Participation

- [X] I am willing to submit a pull request to implement this change.

### Additional comments

[Hashbang Comment](https://tc39.es/ecma262/#prod-HashbangComment) is added since ES2023

### PR DESCRIPTION
**Pull request by sosukesuzuki**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[x] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

Fixes #16913

### CODE REVIEW

**Review by fasttime — CHANGES_REQUESTED** [review]
Thanks! There is a typo in the commit message and in the PR title. There should be an "s" in "afterHa[s]hbangComment".

File: tests/lib/rules/lines-around-comment.js [hunk 1]
```diff
@@ -1051,6 +1051,12 @@ ruleTester.run("lines-around-comment", rule, {
         {
             code: "foo\n/* this is pragmatic */",
             options: [{ applyDefaultIgnorePatterns: false, ignorePattern: "pragma" }]
+        },
+
+        // Hashbang comment
+        {
+            code: "#!comment\n\nvar a = 1;",
+            options: [{ afterHashbangComment: true }]
         }
```
Comment (mdjermanovic) [inline tests/lib/rules/lines-around-comment.js:1060]:
```suggestion
        },
        "#!comment\nvar a = 1;",
        {
            code: "#!comment\nvar a = 1;",
            options: [{}]
        },
        {
            code: "#!comment\nvar a = 1;",
            options: [{ afterHashbangComment: false }]
        },
        {
            code: "#!comment\nvar a = 1;",
            options: [{ afterLineComment: true, afterBlockComment: true }]
        }
```

A few tests to ensure that the rule doesn't require line after hashbang if `afterHashbangComment` isn't set to `true`.

**Review by mdjermanovic — CHANGES_REQUESTED** [review]

**Review by fasttime — APPROVED** [review]
LGTM

File: docs/src/rules/lines-around-comment.md [hunk 2]
```diff
@@ -717,6 +718,35 @@ foo();
...
+
+Examples of **incorrect** code for this rule with the `{ "afterHashbangComment": true }` option:
+
+::: incorrect
+
+```js
+#!foo
+var day = "great"
+
+/*eslint lines-around-comment: ["error", { "afterHashbangComment": true }] */
```
Comment (snitin315) [inline docs/src/rules/lines-around-comment.md:731]: All of our examples have the `/*eslint rule: []*/` comment at the top, will this still work if we move it to the top? I believe the Hashbang comment should be present at the top of the file that's why you have moved it to bottom.

Comment (sosukesuzuki) [reply docs/src/rules/lines-around-comment.md:731]: Hashbang comments in second line will result in a syntax error, so `/*eslint rule: []*/` cannot be put at top.

**Review by snitin315 — APPROVED** [review]
LGTM. Thank you for contributing :star:

I'll let @mdjermanovic do the final review to confirm if his feedback has been addressed.

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

**Pull request merged**